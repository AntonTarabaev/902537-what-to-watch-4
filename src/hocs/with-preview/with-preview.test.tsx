import withPreview from "@root/hocs/with-preview/with-preview";
import * as renderer from "react-test-renderer";

interface Props {
  children: React.ReactNode;
}

const MockComponent = (props: Props) => {
  const {children} = props;

  return (
    <div>
      {children}
    </div>
  );
};

const MockComponentWrapped = withPreview(MockComponent);

it(`withPreview rendered correctly`, () => {
  const tree = renderer.create((
    <MockComponentWrapped
      src={`path`}
      poster={`path`}
    />
  ), {
    createNodeMock() {
      return {};
    }
  }).toJSON();

  expect(tree).toMatchSnapshot();
});
