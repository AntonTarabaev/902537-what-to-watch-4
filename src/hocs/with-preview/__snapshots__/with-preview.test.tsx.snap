// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`withPreview rendered correctly 1`] = `
<div>
  <video />
</div>
`;
